atd/src/annot.ml

```diff
@@ -10,8 +10,20 @@ let error_at loc s =
   failwith (sprintf "%s:\n%s" (Ast.string_of_loc loc) s)
 
 let field ~section ~field l =
-  let open Option.O in
-  List.assoc section l >>= fun (_, l2) -> List.assoc field l2
+  let fieldmatches = List.filter_map (fun (s, (_, fs)) ->
+    if s = section then Some fs else None) l
+    |> List.map (fun fs ->
+      List.filter_map (fun (f, (l, s)) ->
+        if f = field then Some (l, s) else None)
+      fs)
+    |> List.flatten in
+  match fieldmatches with
+  | [fieldmatch] -> Some fieldmatch
+  | (loc, _) :: others -> error_at loc
+    (sprintf "Duplicate annotation %s.%s (also in:\n  %s\n)" section field
+    (List.map (fun (loc, _) -> (Ast.string_of_loc loc)) others
+     |> String.concat ",/n  "))
```

Comment:
This `/n` looks a bit weird. Shouldn't it be a backslash?

Reply:
Absolutely. I'll fix it and complete the fix by tests.

I neglected that pull requests are branch-wise. I originally wanted to have the commits for https://github.com/ahrefs/atd/issues/89 in an extra pull request. This commit slipped in somewhat accidentally. I better separate the two issues.

Reply:
Moved the fixes for https://github.com/ahrefs/atd/issues/89 to a branch of its own, see https://github.com/ahrefs/atd/pull/201.
Backslash is fixed, tests are added.